api/models/weather.py

```diff
@@ -1,11 +1,92 @@
 from schematics.models import Model
 from schematics.types import FloatType, GeoPointType, ModelType, StringType
+from schematics.types.serializable import serializable
+from unidecode import unidecode
+from api.utils.metatypes import EnumMeta
 
 class PhysicalQuantity(Model):
     value = FloatType(required=True)
     unit = StringType(required=True)
 
 
+class Temperature(Model):
+    value = FloatType(required=True)
+
+    class TemperatureMetrics(StringType, metaclass=EnumMeta):
+        CELSIUS = 'CELSIUS'
+        FAHRENHEIT = 'FAHRENHEIT'
+        KELVIN = 'KELVIN'
+
+    @serializable(type=FloatType, serialized_name='value')
+    def get_temperature(self, *args, **kwargs):
+        if hasattr(self, 'context'):
+            temperature = self.context.get('temperature', 'kelvin')
+
+            if temperature.upper() == self.TemperatureMetrics.CELSIUS:
+                value = round(self.value - 273.0, 2) # Converts Kelvin temperature to Celsius
```

Comment:
Maybe instead of rounding to 2 dpp every time you could use a custom type inheriting from `FloatType` which specifies that centrally.